  0%|                                                                                                               | 0/72 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/colligo/REST-PG/src/rest_pg.py", line 428, in <module>
    main()
  File "/home/colligo/REST-PG/src/rest_pg.py", line 417, in main
    trainer.train_rest_pg(args.train_path, args.val_path, args.output_dir)
  File "/home/colligo/REST-PG/src/rest_pg.py", line 304, in train_rest_pg
    self.supervised_fine_tuning(reasoning_data_path, val_path, f"{output_dir}/sft")
  File "/home/colligo/REST-PG/src/rest_pg.py", line 186, in supervised_fine_tuning
    trainer.train()
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/trainer.py", line 2206, in train
    return inner_training_loop(
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/trainer.py", line 2548, in _inner_training_loop
    tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/trainer.py", line 3749, in training_step
    loss = self.compute_loss(model, inputs, num_items_in_batch=num_items_in_batch)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/trainer.py", line 3836, in compute_loss
    outputs = model(**inputs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/parallel/data_parallel.py", line 194, in forward
    outputs = self.parallel_apply(replicas, inputs, module_kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/parallel/data_parallel.py", line 213, in parallel_apply
    return parallel_apply(
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/parallel/parallel_apply.py", line 127, in parallel_apply
    output.reraise()
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/_utils.py", line 750, in reraise
    raise exception
torch.OutOfMemoryError: Caught OutOfMemoryError in replica 0 on device 0.
Original Traceback (most recent call last):
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/parallel/parallel_apply.py", line 97, in _worker
    output = module(*input, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/peft/peft_model.py", line 1845, in forward
    return self.base_model(
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/peft/tuners/tuners_utils.py", line 216, in forward
    return self.model.forward(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/utils/generic.py", line 943, in wrapper
    output = func(self, *args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 553, in forward
    outputs: BaseModelOutputWithPast = self.model(
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/utils/generic.py", line 943, in wrapper
    output = func(self, *args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 441, in forward
    layer_outputs = decoder_layer(
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/modeling_layers.py", line 83, in __call__
    return super().__call__(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 290, in forward
    hidden_states, self_attn_weights = self.self_attn(
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 231, in forward
    query_states = self.q_proj(hidden_states).view(hidden_shape).transpose(1, 2)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/peft/tuners/lora/layer.py", line 769, in forward
    result = result + lora_B(lora_A(dropout(x))) * scaling
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/colligo/miniconda3/envs/rest-pg/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 125, in forward
    return F.linear(input, self.weight, self.bias)
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 5.00 GiB. GPU 0 has a total capacity of 39.38 GiB of which 4.10 GiB is free. Including non-PyTorch memory, this process has 0 bytes memory in use. Of the allocated memory 34.22 GiB is allocated by PyTorch, and 141.08 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
